include/sound/sof/info.h

```diff
@@ -111,7 +111,7 @@ struct sof_ipc_cc_version {
 
```

Comment:
@xiulipan The updated commit is much better -- the examples help. I'm afraid it still doesn't address the questions of why this change is done -- changing the struct here has no impact to ELF file generation, so reasoning is more complex. Also the commit does not answer why it is safe to do this.

Here's my attempt at a commit message that answers these questions:

```
Align struct sof_ipc_cc_version to firmware definition. This struct is used both in IPC messages and embedded as metadata to the FW image. Use of a variable length array has led to errors in firmware tooling and thus the change.

All existing firmware images shipped with old definition have a compiler description field with length less than 24 bytes, so this kernel side change is safe to do.
```

.. it is not needed to go deep into FW ELF generation (that's not relevant to kernel side), but you do need explain why this change needs to be done, and can be done.

Reply:
Thanks @xiulipan ! This now has all the needed info. Let me try to rephrase a bit:
--cut--
Align struct sof_ipc_cc_version to firmware definition in SOF ABI 3.15.0.
  
The struct definition was changed due to errors in FW build. 
The Cadence XCC compiler produces incorrect linkage section sizes,
when a variable length array is used in the compiler version struct.
The firmware definition was changed to a fixed 24 byte compiler description
string. This length covers all released firmware binaries and thus only a minor
ABI change is needed.

As the same structure is used in IPC messages between driver and firmware,  
the kernel needs to be aligned to firmware change.
--cut--

Is above accurate @lgirdwood ?

Reply:
I would worry if for some reason in the future this size would need to be extended firmware side. I'm not sure that extension would work reliably with only a MINOR bump. Also, I feel like the kernel could have a slightly different structure where it would be a variadic array on kernel side but fixed size on firmware side (of course this only works if this structure isn't embedded in others except at the very end). In that case if the size on the firmware side would increase the kernel wouldn't even care. I wonder if this is feasible.

Reply:
@paulstelian97 Ack, increasing the size later is going to be tricky, that's for sure. But, but, for ABI maintenance, I'd rather keep the structs aligned between FW and kernel. I don't really know the history of strings, but it seems many bets are already made in terms of 16 chars should be enough for any compiler name, so a fixed 24 char description, fits to the rest of the design. If we have pressing need in the future, I'd imagine we'd have a separate sof_ipc_cc_version_extended struct added to provide more extenstive information about newer FW versions, and that can be done with a minor bump again (a new field at end of existing IPC, a new section to ELF).